app/livechat/client/views/app/tabbar/visitorHistory.js

```diff
@@ -38,15 +31,38 @@ Template.visitorHistory.helpers({
 Template.visitorHistory.onCreated(function() {
 	const currentData = Template.currentData();
 	this.visitorId = new ReactiveVar();
+	this.isLoading = new ReactiveVar(true);
+	this.history = new ReactiveVar([]);
+	this.offset = new ReactiveVar(0);
+	this.total = new ReactiveVar(0);
 
-	this.autorun(() => {
-		const room = ChatRoom.findOne({ _id: Template.currentData().rid });
+	this.autorun(async () => {
+		const { room } = await APIClient.v1.get(`rooms.info?roomId=${ currentData.rid }`);
 		if (room) {
 			this.visitorId.set(room.v._id);
 		}
 	});
 
-	if (currentData && currentData.rid) {
-		this.loadHistory = this.subscribe('livechat:visitorHistory', { rid: currentData.rid });
-	}
+	this.autorun(async () => {
+		this.isLoading.set(true);
+		const offset = this.offset.get();
+		if (currentData && currentData.rid) {
+			const { history, total } = await APIClient.v1.get(`livechat/visitors.chatHistory/room/${ currentData.rid }/visitor/${ this.visitorId.get() }?count=${ ITEMS_COUNT }&offset=${ offset }`);
```

Comment:
Please, ensure the `this.visitorId.get()` is assigned, otherwise you just need to return.